### DISCUSSION TITLE
Android 端编译pag静态库出错

### DISCUSSION THREAD

**Discussion by dugufly** [Tencent/libpag#1669] [Q&A] [answered]
我想编译一个android 端的静态库， 我关闭了PAG_BUILD_SHARED选项，编译的时候提示这个错误：

```
   $<TARGET_OBJECTS:tgfx>
  Objects of target "tgfx" referenced but is not an OBJECT library.
```

还需要做其他配置么？

**Comment by 0x1306a94** [comment]
按照这里的步骤执行了么[Dependency Management
](https://github.com/Tencent/libpag#dependency-management)

**Reply by dugufly** [reply]
是的，原生的demo是可以正常编译的，改成编译静态库就报错

**Comment by kevingpqi123** [comment, marked as answer]
这里要使用静态库的原因是啥，Android 平台推荐使用动态库，由于 pag 依赖不同的模块，使用静态库会涉及到多个库文件的合并，相对比较麻烦，如果对于静态库的编译机制和 CMake 了解有限，建议使用官方提供的 aar

**Reply by dugufly** [reply]
Android 平台我想通过C++ 接口调用pag的库（我们添加一些其他的业务逻辑，实际就是做一层封装），如果用动态库，访问不了C++的类。

**Reply by kevingpqi123** [reply]
没有你说的这种限制（动态库访问不了 C++ 层的接口），每个版本我们提供了 C++ 层的 API  接口，libpag_*.*.*_include.zip,  so 加载后可以正常使用 C++ 层的接口，公司内部多个业务都是这样使用的，没有任何问题，是你使用的方法有问题
